### PR TITLE
feat(Scripts): decorrelate dev server from dev mode

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
For now, 
* start (dev server) is linked to dev mode
* build is linked to prod mode

There is no way to build in dev mode, or to start the app in prod mode (minified).

**What is the chosen solution to this problem?**
Decorrelate mode (dev/prod) to build/start.
- set the dev server config in common config. It will just be ignored if webpack is called instead of webpack-dev-server
- accept a `--dev` or `--prod` that set the mode
- default mode is still dev for start, and prod for build.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/scripts/webapp/doc/getting-started.md [hunk 1]
```diff
@@ -12,12 +12,19 @@ yarn add --dev @talend/scripts
...
-  "prepublish": "talend-scripts build",
+  "prepublish": "talend-scripts build:prod",
+
+  "build": "talend-scripts build --dev",
+  "build:prod": "talend-scripts build",
   "analyze": "talend-scripts build --env.analyze",
-  "lint:es": "talend-scripts lint:es",
+
+  "start": "talend-scripts start",
+  "start:prod": "talend-scripts start --prod",
```
Comment (frassinier) [inline packages/scripts/webapp/doc/getting-started.md]: As explained by @annilsson here it makes sense.

File: packages/scripts/webapp/preset/config/webpack.config.js [hunk 2]
```diff
@@ -297,5 +297,17 @@ module.exports = ({ getUserConfig, mode }) => {
...
 		},
+		watchOptions: {
+			aggregateTimeout: 300,
+			poll: 1000,
+		},
+		devServer: {
+			port: 3000,
+			stats: 'errors-only',
+			historyApiFallback: true,
+			contentBase: path.join(process.cwd(), '/dist'),
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/scripts/webapp/preset/config/webpack.config.js]: It works if you use /, but in comparison, it fails. So here it's safe but still a mistake, we should not use any separator while using path.join

File: packages/scripts/webapp/doc/getting-started.md [hunk 3]
```diff
@@ -12,12 +12,19 @@ yarn add --dev @talend/scripts
 3. Define the npm scripts.
 ```json
 {
-  "start": "talend-scripts start",
-  "prepublish": "talend-scripts build",
+  "prepublish": "talend-scripts build:prod",
+
+  "build": "talend-scripts build --dev",
+  "build:prod": "talend-scripts build",
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/scripts/webapp/doc/getting-started.md]: It was to stay consistent in the npm scripts, but it's ok for me I'll change it :)

File: packages/scripts/webapp/index.js [hunk 4]
```diff
@@ -7,12 +7,21 @@ const { printSeparator } = require('../common/log');
...
+	} else if (options.includes('--prod')) {
+		userMode = 'production';
+	}
+
+	const restOptions = options.filter(opt => opt !== '--dev' && opt !== '--prod');
 
 	// current env vars and talend scripts configuration in <project-folder>/talend-scripts.json
 	const env = getEnv();
-	env.TALEND_MODE = mode;
+	env.TALEND_MODE = userMode || defaultMode;
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/scripts/webapp/index.js]: done

File: packages/scripts/webapp/preset/config/webpack.config.js [hunk 5]
```diff
@@ -297,5 +297,17 @@ module.exports = ({ getUserConfig, mode }) => {
 			// https://twitter.com/wSokra/status/969679223278505985
 			runtimeChunk: true,
 		},
+		watchOptions: {
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/scripts/webapp/preset/config/webpack.config.js]: This is not the config Karma uses. It has its own webpack config

**Pull request merged**